### ISSUE TITLE
Update the plugin (and the API) structure

### ISSUE THREAD

**Issue by mostaphaRoudsari** [mostaphaRoudsari/Butterfly#84]
Currently, the case gets updated gradually in the process, and it's hard to follow the updates without checking the code. Also adding the recipe once the meshing is done is not how people work with OpenFOAM.
I'm thinking to bring the recipe to an earlier step so once you get a case it's ready to be solved either by butterfly or cmd. It can also potentially create a cleaner folder structure based on recipes where they can all share the mesh if needed.

![updatebfstructure](https://cloud.githubusercontent.com/assets/2915573/19214293/5b0f7d96-8d4d-11e6-8056-f77031e148b1.jpg)

**Comment by kostnermo** [comment]
> I'm thinking to bring the recipe to an earlier step so once you get a case it's ready to be solved either by butterfly or cmd.

I think this is a good idea!

**Comment by mostaphaRoudsari** [comment]
This will also solve current issues for updating fvSchemes and fvSolution. They have to move under recipe and once the case is written they are related to the analysis. Handling them later is possible but not clean. A good example is adding/removing `div(phi,T)		bounded Gauss linearUpwind default;` to/from divSchemes.

**Comment by mostaphaRoudsari** [comment]
Final iteration:
![image](https://cloud.githubusercontent.com/assets/2915573/19422553/455b384e-93e4-11e6-97e1-089a2d65ea82.png)
